mesonbuild/modules/pkgconfig.py

```diff
@@ -328,23 +330,24 @@ def generate_libs_flags(libs):
                 ofile.write('Libs: {}\n'.format(' '.join(generate_libs_flags(deps.pub_libs))))
             if len(deps.priv_libs) > 0:
                 ofile.write('Libs.private: {}\n'.format(' '.join(generate_libs_flags(deps.priv_libs))))
-            ofile.write('Cflags:')
-            for h in subdirs:
-                ofile.write(' ')
-                if h == '.':
-                    ofile.write('-I${includedir}')
-                else:
-                    ofile.write(self._escape(PurePath('-I${includedir}') / h))
-            for f in deps.cflags:
-                ofile.write(' ')
-                ofile.write(self._escape(f))
-            ofile.write('\n')
+
+            def generate_compiler_flags(dirs):
+                cflags_buf = []
+                for f in deps.cflags:
+                    cflags_buf.append(self._escape(f))
+                return cflags_buf
+
+            cflags = generate_compiler_flags(subdirs)
+            if cflags:
```

Comment:
Shouldn't this be conditional on `not dataonly`?

Reply:
I suppose that should be additionally conditional on ```not dataonly```